editor/components/text-with-color-indicators/index.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * External dependencies
+ */
+import { isArray } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import ColorIndicator from '../color-indicator';
+
+const mapColorIndicator = ( settings, index ) => (
+	<ColorIndicator
+		key={ index }
+		colorValue={ settings.value }
+		ariaLabel={ settings.colorIndicatorAriaLabel }
+	/>
+);
+
+const TextWithColorIndicators = ( { className, colorSettings, children } ) => (
```

Comment:
Can we add this component inside the PanelColorSettings and not export it? It is very simple and each component we expose will require maintenance in the future in case we do more changes and we need to deprecate it.